ui/pages/home/home.component.js

```diff
@@ -778,15 +799,87 @@ export default class Home extends PureComponent {
     );
   };
 
+  getIsPrivacyToastRecent() {
+    const { newPrivacyPolicyToastShownDate } = this.props;
+
+    const currentDate = new Date();
+    const oneDayInMilliseconds = 24 * 60 * 60 * 1000;
+    const newPrivacyPolicyToastShownDateObj = new Date(
+      newPrivacyPolicyToastShownDate,
+    );
+    const toastWasShownLessThanADayAgo =
+      currentDate - newPrivacyPolicyToastShownDateObj < oneDayInMilliseconds;
+
+    return toastWasShownLessThanADayAgo;
+  }
+
+  renderToasts() {
+    const { t } = this.context;
+    const {
+      showSurveyToast,
+      showPrivacyPolicyToast,
+      setSurveyLinkLastClickedOrClosed,
+      setNewPrivacyPolicyToastClickedOrClosed,
+    } = this.props;
+    const { theme } = document.documentElement.dataset;
+    const inverseTheme =
+      theme === ThemeType.light ? ThemeType.dark : ThemeType.light;
+
+    const isPrivacyToastRecent = this.getIsPrivacyToastRecent();
+
+    return showSurveyToast || showPrivacyPolicyToast ? (
+      <Box className="home__overlay-banners">
+        {showSurveyToast && (
+          <BannerBase
+            data-theme={inverseTheme}
+            backgroundColor={BackgroundColor.backgroundAlternative}
+            startAccessory={
+              <Icon name={IconName.Heart} color={IconColor.errorDefault} />
+            }
+            title={t('surveyTitle')}
+            actionButtonLabel={t('surveyConversion')}
+            actionButtonOnClick={() => {
+              global.platform.openTab({
+                url: SURVEY_LINK,
+              });
+              setSurveyLinkLastClickedOrClosed(Date.now());
+            }}
+            onClose={() => {
+              setSurveyLinkLastClickedOrClosed(Date.now());
+            }}
+          />
+        )}
+        {showPrivacyPolicyToast && isPrivacyToastRecent && (
```

Comment:
What does the `isPrivacyToastRecent` usage represent?  I know it checks for the last day, but for what reason?

Reply:
This is so that if the user has already been presented with the Toast, but the user has not acted on it, and more than a day has passed, then the toast is not shown anymore to this user.